src/common/generator.js

```diff
@@ -21,9 +21,21 @@ window.POG=(function() {
             else {
                 selector += '[' + name + '=\'' + value + '\']';
             }
-            if (document.querySelectorAll(selector).length === 1) {
-                response = selector;
-            }
+	    //Surrounding with try to catch the exceptions that occur and crash the program on
+	    //certain web pages. 
+	    try
+	    {
+		if (document.querySelectorAll(selector).length === 1) {
+                    response = selector;
+		}
```

Comment:
Got it. I'll integrate it when I get time.

On Friday, 5 August 2016, Bogdan Livadariu <notifications@github.com> wrote:

> In src/common/generator.js
> <https://github.com/rickypc/selenium-page-object-generator/pull/8#discussion_r73717740>
> :
>
> > @@ -21,9 +21,21 @@ window.POG=(function() {
> >              else {
> >                  selector += '[' + name + '=\'' + value + '\']';
> >              }
> > -            if (document.querySelectorAll(selector).length === 1) {
> > -                response = selector;
> > -            }
> > +	    //Surrounding with try to catch the exceptions that occur and crash the program on
> > +	    //certain web pages.
> > +	    try
> > +	    {
> > +		if (document.querySelectorAll(selector).length === 1) {
> > +                    response = selector;
> > +		}
>
> @VaccinalBowl <https://github.com/VaccinalBowl> I was thinking it would
> be nice to throw errors directly at the user using:
> https://github.com/rickypc/selenium-page-object-generator/blob/master/src/
> chrome/assets/js/notify.js
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/rickypc/selenium-page-object-generator/pull/8/files/fb5ce2f9c19c48fcbd93f9af69658f2139303158#r73717740>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAfkjGT8Rem9PBsHSCsdS7hXrOp6R0AYks5qc1-igaJpZM4I22Xs>
> .
>